HadesScheduler/k8s/k8s.go

```diff
@@ -0,0 +1,97 @@
+package k8s
+
+import (
+	"context"
+
+	"github.com/ls1intum/hades/shared/payload"
+	"github.com/ls1intum/hades/shared/utils"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/kubernetes"
+)
+
+type Scheduler struct {
+	// TODO: This may be problematic - We need to clarify how to access the cluster with the service account and find a solution that is compatible with both modes
+	k8sClient *kubernetes.Clientset
+	namespace string
+}
+
+type K8sConfig struct {
+	// K8sNamespace is the namespace in which the jobs should be scheduled (default: hades-executor)
+	// This may change in the future to allow for multiple namespaces
+	K8sNamespace string `env:"K8S_NAMESPACE,notEmpty" envDefault:"hades-executor"`
+
+	// K8sConfigMode is used to determine how the Kubernetes client should be configured ("kubeconfig" or "serviceaccount")
+	ConfigMode string `env:"K8S_CONFIG_MODE,notEmpty" envDefault:"kubeconfig"`
+}
+
+// K8sConfigKubeconfig is used as configuration if used with a kubeconfig file
+type K8sConfigKubeconfig struct {
+	kubeconfig string `env:"KUBECONFIG"`
+}
+
+// K8sConfigServiceaccount is used as configuration if used with a service account
+type K8sConfigServiceaccount struct {
+}
```

Comment:
Personally, I prefer to have tested structs in this case. I mean in theory we can check for all `env` variables and try to load them. Depending on the `ConfigMode` we then select one of the nested structs